File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.ts
/** 
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * var guess = function(num) {}
 */


function guessNumber(n: number): number {
    let start = 1;
    let end = n;

    while (start <= end) {
       let currNum = Math.floor((end + start) / 2);
        if (guess(currNum) === -1) {
            end = currNum - 1;
        } else if (guess(currNum) === 1){
            start = currNum + 1;
        } else {
            return currNum;
        }
    }
    return -1
};

File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        s = 0
        e = n

        while s <= e:
            currentNum = round(s + (e - s) / 2)
            result = guess(currentNum)

            if result > 0:
                s = currentNum + 1

            elif (result < 0):
                e = currentNum - 1
            else:
                return currentNum